router/batchrouter/asyncdestinationmanager/snowpipestreaming/snowpipestreaming.go

```diff
@@ -513,6 +514,18 @@ func (m *Manager) Poll(pollInput common.AsyncPoll) common.PollStatusResponse {
 
 	if anyInProgress := m.processPollImportInfos(ctx, importInfos); anyInProgress {
 		m.stats.pollingInProgress.Increment()
+
+		// Check for stuck pipeline batch
+		now := m.Now()
+		if m.pollingStartTime.IsZero() {
+			m.pollingStartTime = now
+		} else {
+			duration := now.Sub(m.pollingStartTime)
+			threshold := m.config.stuckPipelineThreshold.Load()
+			if duration > threshold {
+				m.logger.Warnn("Stuck snowipe pipeline detected")
```

Comment:
Should we add additional information for debugging, like importID?

Reply:
Added some fields